### ISSUE TITLE
Blurry Results?

### ISSUE THREAD

**Issue by thekesslereffect** [lucidrains/stylegan2-pytorch#160]
I have about 96,000 iterations so far and my results are still quite blurry. Is there a specific variable I should look at to get more detail? Or do I just need to train it longer? Overall the generation is looking really good, it just doesn't have the nice detail of the original data.

These are my current settings for my rtx 2060 6Gb:
input data 128px
image-size 128px
batch-size 2
gradient-accumulate-every 5
network-capacity 8
aug-prob 0.25 (I only have 760 input data images, but they are all almost the same, just variations in color and details. So, maybe I can change this back to default?)
top-k-training is on

**Comment by lucidrains** [comment]
@thekesslereffect if it hasn't collapsed yet, then training isn't finished, just keep training! after that, the only option you have is to scale up the resolutions

your batch size is also way too tiny, you want at least effective batch size of 32 (batch size * gradient accumulate every). yours is currently 10

network capacity is also quite small, but maybe it's ok for you, since you don't have much data anyhow!

**Comment by lucidrains** [comment]
@thekesslereffect also, turn on attention :)

**Comment by lucidrains** [comment]
@thekesslereffect you should be happy to see any results at all, since 760 images is not a lot by machine learning standards

**Comment by ZhangYuef** [comment]
> @thekesslereffect if it hasn't collapsed yet, then training isn't finished, just keep training! after that, the only option you have is to scale up the resolutions
> 
> your batch size is also way too tiny, you want at least effective batch size of at least 32 (batch size * gradient accumulate every). yours is currently 10
> 
> network capacity is also quite small, but maybe it's ok for you, since you don't have much data anyhow!

What's the meaning of collapse here?

**Comment by Dhruva-Storz** [comment]
Probably a bit late, but ill put it out there anyway. He probably means training collapse where the minimax game between the discriminator and generator collapses, which is observed by the loss curves of D and G diverging from one another. 

In case you weren't aware, collapse in the context of GANs could also mean mode collapse, where the generator learns that it can fool the discriminator with a specific type of image, which manifests as many of the images from different parts of the latent space looking the same. This is also a fail state in GAN training as it means something went wrong.
